modules/logs/classes/class-log-install.php

```
@@ -147,8 +152,63 @@ public function install() {
+        if ( ! empty( $currentVersion ) && version_compare( $currentVersion, '1.0.1.20', '<' ) ) { // NOSONAR - non-ip.
+            $this->wpdb->query( 'ALTER TABLE ' . $this->table_name( 'wp_logs_archive' ) . ' ADD COLUMN user_login varchar(100) NOT NULL' ); //phpcs:ignore -- ok.
+            $this->wpdb->query( 'ALTER TABLE ' . $this->table_name( 'wp_logs_archive' ) . ' ADD INDEX user_login ( user_login )' ); //phpcs:ignore -- ok.
+        }
+
+        if ( ! empty( $currentVersion ) && version_compare( $currentVersion, '1.0.1.26', '<' ) ) { // NOSONAR - non-ip.
+            $count = Log_DB_Helper::instance()->count_legacy_dismissed();
+            if ( ! empty( $count ) ) {
+                update_option( 'mainwp_module_logs_updates_dismissed_db_process_status', 'require_update' );
+            }
+        }
+    }
```

PHPMD (2.15.0)

The method update_log_db() has a Cyclomatic Complexity of 14. The configured cyclomatic complexity threshold is 10. (Code Size Rules)

(CyclomaticComplexity)